lxd/instance/drivers/driver_qemu.go

```diff
@@ -5129,36 +5139,14 @@ func (d *qemu) Update(args db.InstanceArgs, userRequested bool) error {
 				return d.architectureSupportsCPUHotplug()
 			}
 
-			if strings.HasPrefix(key, "boot.") {
-				return true
-			}
-
-			if strings.HasPrefix(key, "cloud-init.") {
-				return true
-			}
-
-			if strings.HasPrefix(key, "environment.") {
-				return true
-			}
-
-			if strings.HasPrefix(key, "image.") {
-				return true
-			}
-
-			if strings.HasPrefix(key, "snapshots.") {
-				return true
-			}
-
-			if strings.HasPrefix(key, "user.") {
-				return true
-			}
-
-			if strings.HasPrefix(key, "volatile.") {
+			if shared.StringInSlice(key, liveUpdateKeys) {
 				return true
 			}
 
-			if shared.StringInSlice(key, liveUpdateKeys) {
-				return true
+			for _, prefix := range liveUpdateKeyPrefixes {
```

Comment:
You can use `shared.StringHasPrefix` here.

Reply:
Thanks, much cleaner :) Changed.